yt_dlp/extractor/tver.py

```diff
@@ -171,7 +166,87 @@ def _real_extract(self, url):
             'channel': provider,
             'description': str_or_none(video_info.get('description')),
             'thumbnails': thumbnails,
+        }
+
+        backend = self._configuration_arg('backend', ['streaks'])[0]
+
+        if backend not in ('brightcove', 'streaks'):
+            raise ExtractorError(f'Invalid backend value: {backend}', expected=True)
+
+        if backend == 'brightcove':
+            return self._brightcove_backend(data, video_info)
+
+        return self._streaks_backend(data, video_info, video_id)
+
+    def _brightcove_backend(self, result, video_info):
+        self.write_debug('Using Brightcove backend')
+
+        p_id = video_info['video']['accountID']
+        r_id = traverse_obj(video_info, ('video', ('videoRefID', 'videoID')), get_all=False)
+
+        if not r_id:
+            raise ExtractorError('Failed to extract reference ID for Brightcove')
+
+        if not r_id.isdigit():
+            r_id = f'ref:{r_id}'
+
+        result.update({
+            '_type': 'url_transparent',
             'url': smuggle_url(
                 self.BRIGHTCOVE_URL_TEMPLATE % (p_id, r_id), {'geo_countries': ['JP']}),
             'ie_key': 'BrightcoveNew',
-        }
+        })
+
+        return result
+
+    def _streaks_backend(self, result, video_info, video_id):
+        self.write_debug('Using streaks.jp backend')
+
+        ref_id = traverse_obj(video_info, ('streaks', 'videoRefID'), get_all=False)
+        project_id = traverse_obj(video_info, ('streaks', 'projectID'), get_all=False)
+
+        if not ref_id:
+            raise ExtractorError('Failed to extract reference ID for streaks.jp stream info')
+
+        if not project_id:
+            raise ExtractorError('Failed to extract project ID for streaks.jp stream info')
+
+        if not ref_id.isdigit() and not ref_id.startswith('ref:'):
```

Comment:
This looks not correct. 
For streak backend , all ref_id will prepend "ref:" , no matter it is digits or not. For example: https://tver.jp/episodes/epwkfr091w

Reply:
Yeah, i actually just copied what brightcove extractor was doing, i discovered today that indeed you need ref:(digit) as well. i have pushed a fix for it now,